syntax = "proto2";

package console_server_message;

message total_message {
    required uint32 type = 1;
    optional uint32 key = 2;
    optional string value = 3;
    optional uint32 count = 4;
    optional string error = 5;
    optional string flag = 6;
}

message r7_msg {
    required uint32 type = 1;
    required uint32 count = 2;
}

message r4_msg {
    required uint32 type = 1;
    required string value = 2;
}

message r5_msg {
    required uint32 type = 1;
}

message r8_msg {
    required uint32 type = 1;
    required string error = 2;
}

message flag {
    required uint32 type = 1;
    required string flag = 2;
}

